Week of May 12 (#348)
xregistry/spec: core/spec.md

  value references the correct type of Registry entity. See the definition of
  the [`target` model attribute](#model.target) for more information.
  Its value MUST start with a `/`.
- `xidtype` - MUST be a `URL` reference to an xRegistry model type. The

>    reference MUST point to oe of: the Registry itself (`/`), a Group type

Reviewer: ```suggestion
   reference MUST point to one of: the Registry itself (`/`), a Group type
```

Author: done

In order for a client to discover the list of available values for each
capability, an HTTP `GET` MAY be sent to the `/capabilities` API with the
`?offered` query parameter and the response MUST adhere to the following

> (which borrows many of the same structure from the model definition language):

Reviewer: "many of the same structures" or "much of the same structure" perhaps?

#### Retrieving the Registry Model

The Registry model is available in two forms:
- The full "model" with all possible aspects of the model defined.
- The "modelsource" form represents just the model aspects specified when
  the model was defined.

The full "model" view can be thought of as a full schema definition of what the

> messages exchanges with the server might look like. As such, it'll include

Reviewer: ```suggestion
messages exchanges with the server might look like. As such, it includes
```

Reviewer: (Here and below, the future tense feels a bit odd.)

Author: done - both

the Groups, Resources, model-specific attributes, extension attributes,
specification defined attributes and overrides to those specification defined
attributes.

The "modelsource" view of the model is just what was provided by the user when
the model was defined, or last edited. It is expected that this view of the

> model will be much smaller than the full model and will only include

Reviewer: ```suggestion
model is much smaller than the full model and only includes
```

domain-specific information. While specification defined attributes MAY appear
in this document, they are NOT RECOMMENDED since the server will automatically
add them and so users do not need to concern themselves with those details.

> The modelsource document will always be a semantic subset of the full model

Reviewer: ```suggestion
The modelsource document is always a semantic subset of the full model
```

Author: done

     value).
    - When `VALUE` is non-`null` then only entities without the specified
      `ATTRIBUTE` and `VALUE` MUST be included in the response. This MUST be
      semantically equivalent to `NOT(ATTRIBUTE=VALUE)`, and this also means
      that if `ATTRIBUTE` is missing then that attribute will match the filter.
  - `<`, `<=`, `>`, `>=` operators:
    - `VALUE` MUST NOT be `null`.
    - Only entities with the specified `ATTRIBUTE` and `VALUE` that matches
      the comparison operator's semantics MUST be included in the response.
    - `<` refers to "less than".
    - `<=` refers to "less than or equal to".
    - `>` refers to "greater than".
    - `>=` refers to "greater than or equal to".
    - Wildcards (`*`) (see below) MUST NOT be present in the `VALUE`.
    - MUST only be used on numeric or string attribute types.

When comparing an `ATTRIBUTE` to the specified `VALUE`, the type of the
attribute impacts how the comparisons are done:
- For boolean attributes, a successful `=` comparison MUST be an exact
  case-sensitive match (`true` or `false`).
- For numeric attributes, standard numeric comparisons rules apply.

> - For string attributes, the strings MUST be compared in a case-insensitive

Reviewer: String *values* can be any Unicode, I assume? In which case we need to be more specific about comparisons. Ordinal, culture-insensitive, something else?

Reviewer: I'd need to know what we actually want to happen first :)
